### DISCUSSION TITLE
简单条形图y轴刻度设置最小值，图形渲染异常

### DISCUSSION THREAD

**Discussion by Calmio-Y** [antvis/G2#6188] [Q&A]
### 问题描述

import { Chart } from '@antv/g2';

const data = [
  { year: '1951 年', sales: 38 },
  { year: '1952 年', sales: 52 },
  { year: '1956 年', sales: 61 },
  { year: '1957 年', sales: 145 },
  { year: '1958 年', sales: 48 },
  { year: '1959 年', sales: 38 },
  { year: '1960 年', sales: 38 },
  { year: '1962 年', sales: 38 },
];

const chart = new Chart({
  container: 'container',
  autoFit: true,
});

chart
  .interval()
  .coordinate({ transform: [{ type: 'transpose' }] })
  .data(data)
  .encode('x', 'year')
  .encode('y', 'sales')
   .scale("y", {
    domainMin: 30, 
    nice: true, 
  })

chart.render();

### 预期行为

设置刻度最值时，图形正常渲染

### 平台

- 操作系统: [macOS, Windows, Linux, React Native ...]
- 网页浏览器: [Google Chrome, Safari, Firefox]

### 屏幕截图或视频（可选）

<img width="929" alt="22" src="https://github.com/antvis/G2/assets/30626950/bc6817ba-082f-45e6-b6cf-8acc29c04095">

**Comment by pearmini** [comment]
这个需要设置 chart 的 clip 属性去裁剪超出的部分：

```js
import { Chart } from '@antv/g2';

const data = [
  { year: '1951 年', sales: 38 },
  { year: '1952 年', sales: 52 },
  { year: '1956 年', sales: 61 },
  { year: '1957 年', sales: 145 },
  { year: '1958 年', sales: 48 },
  { year: '1959 年', sales: 38 },
  { year: '1960 年', sales: 38 },
  { year: '1962 年', sales: 38 },
];

const chart = new Chart({
  container: 'container',
  autoFit: true,
  clip: true,
});

chart
  .interval()
  .coordinate({ transform: [{ type: 'transpose' }] })
  .data(data)
  .encode('x', 'year')
  .encode('y', 'sales')
  .scale('y', {
    domainMin: 30,
    nice: true,
  });

chart.render();
```

**Reply by Calmio-Y** [reply]
![image](https://github.com/antvis/G2/assets/30626950/4f28397f-9127-4f01-a9a6-2a86918502cc)
加完clip 属性，会出现上述图片的问题

**Reply by Calmio-Y** [reply]
6191的图例完全是G2绘制出来的。使用方并没有绘制图例。怎么写自定义函数去更新

**Reply by pearmini** [reply]
> 加完clip 属性，会出现上述图片的问题

那使用 clamp 属性：

```js
import { Chart } from '@antv/g2';

const data = [
  { year: '1951 年', sales: 38 },
  { year: '1952 年', sales: 52 },
  { year: '1956 年', sales: 61 },
  { year: '1957 年', sales: 145 },
  { year: '1958 年', sales: 48 },
  { year: '1959 年', sales: 38 },
  { year: '1960 年', sales: 38 },
  { year: '1962 年', sales: 38 },
];

const chart = new Chart({
  container: 'container',
  autoFit: true,
});

chart
  .interval()
  .coordinate({ transform: [{ type: 'transpose' }] })
  .data(data)
  .encode('x', 'year')
  .encode('y', 'sales')
  .scale('y', {
    domainMin: 30,
    nice: true,
    clamp: true
  });

chart.render();
```

**Reply by pearmini** [reply]
> 6191的图例完全是G2绘制出来的。使用方并没有绘制图例。怎么写自定义函数去更新

刚才链接贴错了，在这里：https://github.com/antvis/G2/discussions/6184 回复了

**Comment by Calmio-Y** [comment]
![image](https://github.com/antvis/G2/assets/30626950/f952b623-ad6d-484a-bf98-1ed50efb12da)
加上clamp属性
这是符合预期的吗
